File: BHAVRP_Java/src/cujae/inf/citi/om/factory/interfaces/IFactoryAssignment.java
package cujae.inf.citi.om.factory.interfaces;

import cujae.inf.citi.om.heuristic.assignment.Assignment;

/* Interfaz que define cómo crear un objeto Assignment*/

public interface IFactoryAssignment {
	
	public Assignment createAssignment(AssignmentType assignmentType);
}

File: BHAVRP_Python/cujae/inf/citi/om/factory/interfaces/ifactory_assignment.py
from abc import ABC, abstractmethod
from .assignment_type import AssignmentType
from ...assignment.assignment import Assignment

# Interfaz que define como crear un objeto Assignment.
class IFactoryAssignment(ABC):
    
    @abstractmethod
    def create_assignment(self, assignment_type: AssignmentType) -> Assignment:
        pass